### ISSUE TITLE
Add a setting to show pictures of rooms and spaces.

### ISSUE THREAD

**Issue by zarp-dev** [ajbura/cinny#107]
#### Is your feature request related to a problem? Please describe.
It would be great to have a setting to show pictures of rooms and spaces.
![Screenshot 2021-09-18 at 14 32 35](https://user-images.githubusercontent.com/65812041/133887489-5e18c76d-606b-487e-b15b-4a1fca202f13.png)

#### Describe the solution you'd like
![Screenshot 2021-09-18 at 14 38 31](https://user-images.githubusercontent.com/65812041/133887536-328e88c6-f206-48e1-80c4-528b9f2aa153.png)

**Comment by kfiven** [comment]
Blend of Pictures and Icons looks ugly as well as confusing. imho either we should keep all icons or all pictures. 

Also room icon is visible after opening the room so I don't get why we need to duplicate it in drawer.

**Comment by Salamandar** [comment]
Bump ! This is one of the few last things I prefer on Element ^^''

> Also room icon is visible after opening the room so I don't get why we need to duplicate it in drawer.

Uh… Maybe because pictures are here to recognize/differentiate multiple rooms ?

**Comment by gabrc52** [comment]
> Blend of Pictures and Icons looks ugly as well as confusing. imho either we should keep all icons or all pictures.

I agree but I would like this feature too, but the rooms in the proposed solution image dont look very clean to be honest. Spaces look great tho. This app has great design, which we don't want to break. Maybe all pictures (optional perhaps) would work

> Also room icon is visible after opening the room so I don't get why we need to duplicate it in drawer.

What it would achieve: It gives you a birds eye view of the rooms you have. The room you're looking for catches your eye so you can click on it faster or find it faster.

**Comment by Salamandar** [comment]
> Maybe all pictures (optional perhaps) would work

Yes, agreed. This could be a setting to let the user choose.

**Comment by Salamandar** [comment]
> Blend of Pictures and Icons looks ugly as well as confusing.

…That's already what we have for people. Do you find Cinny ugly and confusing for that ?

**Comment by Salamandar** [comment]
I've wrote #1076 that successfully fixes this issue. It doesn't let the user choose though, maybe @kfiven would like to contribute a setting for this ?

**Comment by kfiven** [comment]
> > Blend of Pictures and Icons looks ugly as well as confusing.
> 
> …That's already what we have for people. Do you find Cinny ugly and confusing for that ?

I meant a blend of icons and pictures in one list. That's why I said either we should have icons or just pictures.

**Comment by Salamandar** [comment]
Yes, i understood. In the people list, you might have their avatar or a color circle with a letter if they don't have an avatar. So we already have mixed display in Cinny.

**Comment by belkka** [comment]
I'll duplicate what I wrote in #1076:

Today I came to github only to fill a feature request on topic... but found this PR. I believe group avatars should be visible (isn't it their purpose?); it's so common and familiar for ~my friends~ potential users migrating from other popular messengers.

Current UI reminds channels in slack/discord spaces though — is it an actual motivation? In that case I do agree it's a nice feature, but I feel that:
- it's only reasonable **in spaces, not in global "home" list**. Spaces do feel like slack/discord servers and UI with no room avatars could be fine for them because
  - spaces/servers already have icons
  - there are usually not much rooms within single space/server (easier to navigate) and they share common theme & community (lower risks of choosing wrong room)
  - having clean screen is also good (BTW, some discord servers use emoji in room names to compensate absence of icons)

  But "home" space feels more like a usual "chat-oriented" (rather than "space-oriented") messenger (Telegram, WhatsApp...) — a place where divergent uncategorized chats (communities) are shuffled together. This factor makes visual distinction **more valuable**. I'd say rooms outside of spaces are kind of "mini spaces" themself.
- despite what has been said above, even with this PR being merged, space admins **will be able** to imitate slack/discord's look and feel if they like to... simply by not setting any room avatars and therefore keeping default hashtag icons!

I don't really want my matrix client hiding room avatars from me. And I really like Cinny, wish I could suggest it to my friends as a feature-rich alternative to messaging apps that do not respect privacy.

**Comment by ajbura** [comment]
I closed the #2303 as per https://github.com/cinnyapp/cinny/pull/2303#pullrequestreview-2971866386 but will try thinking through designs if we can somehow bring room avatar without loosing current experience.
